gen/build_deploy/bash.py

```diff
@@ -669,24 +668,35 @@ def fill_template(base_name, format_args):
     return installer_filename
 
 
-def make_dcos_launch():
-    # NOTE: this needs to be kept in sync with build_dcos_launch.sh
+def get_launch_package_id(package_list):
+    """scans througha the package list of the
+    """
+    launch_package_id = None
+    for package_id in package_list:
+        if pkgpanda.PackageId(package_id).name == 'dcos-launch':
+            launch_package_id = package_id
+            break
+    assert launch_package_id, 'dcos-launch package cannot be found!'
+    return launch_package_id
+
+
+def fetch_dcos_launch_bin(launch_name, launch_package_id):
     work_dir = py.path.local.mkdtemp()
-    # the tree will not be in the working dir if run from downstream
-    tree_src = './'
-    if os.path.exists('ext/upstream'):
-        tree_src = './ext/upstream'
-    src_info = {'kind': 'git_local', 'rel_path': tree_src}
-    this_tree = GitLocalSrcFetcher(src_info, None, os.getcwd())
-    this_tree.checkout_to(str(work_dir))
-    # put the launch spec into the root of the tree before running pyinstaller
-    work_dir.join('dcos-launch.spec').write(pkg_resources.resource_string('launch', 'dcos-launch.spec'))
-    with work_dir.as_cwd():
-        subprocess.check_call(['pyinstaller', '--log-level=DEBUG', 'dcos-launch.spec'])
-    subprocess.check_call(['mv', str(work_dir.join('dist').join('dcos-launch')), 'dcos-launch'])
+    # inspect the local packages cache to grab the desired artifact
+    cached_package_path = 'packages/cache/packages/{pkg_name}/{pkg_id}.tar.xz'.format(
+        pkg_name=pkgpanda.PackageId.parse(launch_package_id)[0], pkg_id=launch_package_id)
+    subprocess.check_call(['tar', '-Jxf', cached_package_path, '-C', str(work_dir)])
+    launch_path = 'packages/cache/' + launch_name
+    subprocess.check_call(['cp', str(work_dir.join('dcos-launch')), launch_path])
     work_dir.remove()
+    return launch_path
+
 
-    return "dcos-launch"
+def get_dcos_launch_name(variant):
+    if variant is None:
+        return 'dcos-launch'
+    else:
+        return 'dcos-launch' + pkgpanda.util.variant_suffix(variant, delim='-')
```

Comment:
No need for this function now that we have `variant_suffix()`. That'll do the work of returning an empty suffix `if variant is None`.